server/lib/api/users.js

```diff
@@ -97,3 +106,33 @@ async function resetPassword(ctx, next) {
     ctx.status = 204
   })
 }
+
+async function verifyEmail(ctx, next) {
```

Comment:
And in cases where we limit on IP and not user, you'd want to consider what effects multiple machines behind the same NAT would have (but for this particular case, limiting on user+email combo seems fine tbh)